src/core/ext/filters/client_channel/lb_policy/xds/xds_cluster_resolver.cc

```diff
@@ -510,6 +489,35 @@ void XdsClusterResolverLb::LogicalDNSDiscoveryMechanism::ResolverResultHandler::
       discovery_mechanism_->index(), std::move(update));
 }
 
+//
+// XdsClusterResolverLb::DiscoveryMechanismEntry
+//
+
+const XdsClusterResolverLbConfig::DiscoveryMechanism&
+XdsClusterResolverLb::DiscoveryMechanismEntry::config() const {
+  return discovery_mechanism->parent()
+      ->config_->discovery_mechanisms()[discovery_mechanism->index()];
+}
+
+std::string XdsClusterResolverLb::DiscoveryMechanismEntry::GetChildPolicyName(
+    size_t priority) const {
+  const size_t child_number = priority_child_numbers[priority];
+  const auto& discovery_config = config();
+  absl::InlinedVector<absl::string_view, 9> parts = {
```

Comment:
should this be 10 instead of 9?

Reply:
Actually, it should have been 7.  But this has gone away based on the change I made in response to your other comment.